Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -43,13 +43,28 @@ std::map<std::string, FieldValue> DecodeObject(pb_istream_t* stream);
 
 /**
  * Docs TODO(rsgowman). But currently, this just wraps the underlying nanopb
- * pb_ostream_t. Eventually, this might use static factory methods to create the
- * underlying pb_ostream_t rather than directly passing it in.
+ * pb_ostream_t.
  */
 // TODO(rsgowman): Encode* -> Write*
 class Writer {
  public:
-  explicit Writer(pb_ostream_t* stream) : stream_(stream) {
+  /**
+   * Creates an output stream that writes to the specified vector. Note that
+   * this vector pointer must remain valid for the lifetime of this Writer.
+   *
+   * (This is roughly equivalent to the nanopb function
+   * pb_ostream_from_buffer())
+   *
+   * @param out_bytes where the output should be serialized to.
+   */
+  static Writer FromBuffer(std::vector<uint8_t>* out_bytes);
```

Comment:
Very optional nit: is the static method used for symmetry with `SizingStream`? In that case, maybe a more symmetrical name? Or maybe just make it a constructor (`SizingStream` is kind of a special case)?

Reply:
Also, this name implies a copy of the buffer but you're actually writing *into* the buffer. The direction of the preposition should go the other way. Maybe `static Writer Wrap(std::vector<uint8_t>* out_bytes)`?